aws/resource_aws_eks_node_group_test.go

```diff
@@ -16,65 +16,76 @@ import (
 )
 
 func init() {
-	resource.AddTestSweepers("aws_eks_fargate_node_group", &resource.Sweeper{
-		Name: "aws_eks_fargate_node_group",
-		F:    testSweepEksFargateNodegroups,
+	resource.AddTestSweepers("aws_eks_node_group", &resource.Sweeper{
+		Name: "aws_eks_node_group",
+		F:    testSweepEksNodeGroups,
 	})
 }
 
-func testSweepEksFargateNodegroups(region string) error {
+func testSweepEksNodeGroups(region string) error {
 	client, err := sharedClientForRegion(region)
+
 	if err != nil {
-		return fmt.Errorf("error getting client: %s", err)
+		return fmt.Errorf("error getting client: %w", err)
 	}
+
 	conn := client.(*AWSClient).eksconn
+	sweepResources := make([]*testSweepResource, 0)
+	var errors *multierror.Error
 
-	var errors error
 	input := &eks.ListClustersInput{}
+
 	err = conn.ListClustersPages(input, func(page *eks.ListClustersOutput, lastPage bool) bool {
 		for _, cluster := range page.Clusters {
 			clusterName := aws.StringValue(cluster)
 			input := &eks.ListNodegroupsInput{
 				ClusterName: cluster,
 			}
+
 			err := conn.ListNodegroupsPages(input, func(page *eks.ListNodegroupsOutput, lastPage bool) bool {
-				for _, nodegroup := range page.Nodegroups {
-					nodegroupName := aws.StringValue(nodegroup)
-					log.Printf("[INFO] Deleting EKS Node Group %q", nodegroupName)
-					input := &eks.DeleteNodegroupInput{
-						ClusterName:   cluster,
-						NodegroupName: nodegroup,
-					}
-					_, err := conn.DeleteNodegroup(input)
-
-					if err != nil && !isAWSErr(err, eks.ErrCodeResourceNotFoundException, "") {
-						errors = multierror.Append(errors, fmt.Errorf("error deleting EKS Node Group %q: %w", nodegroupName, err))
-						continue
-					}
-
-					if err := waitForEksNodeGroupDeletion(conn, clusterName, nodegroupName, 10*time.Minute); err != nil {
```

Comment:
This wait is already included when using the resource to delete instead of directly through API.